echopype/convert/parse_azfp.py

```diff
@@ -127,25 +90,36 @@ def load_AZFP_xml(self):
         """Parses the AZFP  XML file.
         """
 
-        def get_value_by_tag_name(tag_name, element=0):
-            """Returns the value in an XML tag given the tag name and the number of occurrences."""
-            return px.getElementsByTagName(tag_name)[element].childNodes[0].data
-
         xmlmap = fsspec.get_mapper(self.xml_path, **self.storage_options)
-        px = xml.dom.minidom.parse(xmlmap.fs.open(xmlmap.root))
-
-        # Retrieve integer parameters from the xml file
-        for old_name, new_name in XML_INT_PARAMS.items():
-            self.parameters[new_name] = int(get_value_by_tag_name(old_name))
-        # Retrieve floating point parameters from the xml file
-        for param in XML_FLOAT_PARAMS:
-            self.parameters[param] = float(get_value_by_tag_name(param))
-        # Retrieve frequency dependent parameters from the xml file
-        for old_name, new_name in XML_FREQ_PARAMS.items():
-            self.parameters[new_name] = [
-                float(get_value_by_tag_name(old_name, ch))
-                for ch in range(self.parameters["num_freq"])
-            ]
+        root = ET.parse(xmlmap.fs.open(xmlmap.root)).getroot()
+
+        for child in root.iter():
+            if len(child.attrib) > 0:
+                for key, val in child.attrib.items():
+                    self.parameters[child.tag + key].append(val)
+
+            if all(char == "\n" for char in child.text):
+                continue
+            else:
+                try:
+                    val = int(child.text)
+                except ValueError:
+                    val = float(child.text)
+
+            if len(child.tag) > 3 and not child.tag.startswith("VTX"):
+                words = re.findall("[A-Z]+[a-z]*", child.tag)
+                words_lower = [word.lower() for word in words]
+                if len(words) > 1:
+                    self.parameters["_".join(words_lower)].append(val)
+                elif len(words) == 1:
+                    self.parameters[words_lower[0]].append(val)
```

Comment:
you could probably wrap these in a subfunction so that you can use here and also in the above for the `child.tag + key.capitalized` fields ([this comment](https://github.com/OSOceanAcoustics/echopype/pull/1135/files?diff=unified&w=0#r1328983167)).